pyslim/slim_tree_sequence.py

```
@@ -490,6 +490,139 @@ def recapitate(self,
+        :param int Ne: The effective population size for :meth:`msprime.simulate`,
+            in units of diploid individuals
+        :param int samples: Sample count for :meth:`msprime.simulate`,
+            in units of haploid samples,
+        :param float mutation_rate: A (constant) mutation rate,
+            in units of mutations per nucleotide per unit of time.
+        :param float recombination_rate: A (constant) recombination rate,
+            in units of crossovers per nucleotide per unit of time.
+        :type recombination_map: :class`msprime.RecombinationMap`
+        :param recombination_map: The recombination map, or None,
+            if recombination_rate is specified.
+        :param dict kwargs: Any other arguments to :meth:`msprime.simulate`.
+        '''
+        if "keep_first_generation" in kwargs:
+            raise ValueError("The keep_first_generation argument is deprecated:"
+                             "the FIRST_GEN flag is no longer used.")
+
+        discrete_msprime = hasattr(msprime, "RateMap")
+
+        if recombination_rate is not None:
+            if recombination_map is not None:
+                raise ValueError("Cannot specify length/recombination_rate along with a recombination map")
+            if discrete_msprime:
+                recombination_map = msprime.RecombinationMap(positions = [0.0, self.sequence_length],
+                                                             rates = [recombination_rate, 0.0])
+            else:
+                recombination_map = msprime.RecombinationMap(positions = [0.0, self.sequence_length],
+                                                             rates = [recombination_rate, 0.0],
+                                                             num_loci = int(self.sequence_length))
+        if discrete_msprime and ('discrete_genome' not in kwargs):
+            kwargs['discrete_genome'] = True
+
+        slim_indivs = self.individuals_alive_at(0)
```

hm, if this becomes an msprime method we'll have to include another argument that says which nodes are the ones "alive at the end" (which could default to the samples)

---

Yes this may be the only pyslim specific line. Wouldn't be too tough to implement though!